pkg/utils/utils_virtual.go

```diff
@@ -205,6 +223,20 @@ func getOpenstackDataFromMetadataService() (metaData *OSPMetaData, networkData *
 	return metaData, networkData, nil
 }
 
+// getPCIAddressFromMACAddress returns the PCI address of a device given its MAC address
+func getPCIAddressFromMACAddress(macAddress string) (string, error) {
+	nics, err := ghw.Network()
+	if err != nil {
+		return "", fmt.Errorf("error getting network info: %v", err)
+	}
+	for _, nic := range nics.NICs {
+		if nic.MacAddress == macAddress {
+			return *nic.PCIAddress, nil
```

Comment:
This will return on the first match. What do we do if there are two devices with the same MAC address? Apparently this is a thing (per @SeanMooney)